pkg/kubectl/cmd/run.go

```diff
@@ -149,64 +159,20 @@ func Run(f *cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *cob
 
 	params["env"] = cmdutil.GetFlagStringSlice(cmd, "env")
 
-	err = kubectl.ValidateParams(names, params)
-	if err != nil {
-		return err
-	}
-
-	obj, err := generator.Generate(params)
-	if err != nil {
-		return err
-	}
-
-	mapper, typer := f.Object()
-	version, kind, err := typer.ObjectVersionAndKind(obj)
-	if err != nil {
-		return err
-	}
-
-	inline := cmdutil.GetFlagString(cmd, "overrides")
-	if len(inline) > 0 {
-		obj, err = cmdutil.Merge(obj, inline, kind)
-		if err != nil {
+	if cmdutil.GetFlagBool(cmd, "expose") {
```

Comment:
I think it would be more appropriate to first generate the container for the new image and then if successful, and `--expose` is true, create the service. I also don't see how this is interplaying with dry-run.